util/atomic_bool.go

```diff
@@ -0,0 +1,39 @@
+package util
+
+import "sync/atomic"
+
+const (
+	falseValue int32 = 0
+	trueValue  int32 = 1
+)
+
+type AtomicBool struct {
+	value int32
+}
+
+func NewAtomicBool(val bool) *AtomicBool {
+	a := &AtomicBool{value: falseValue}
+	a.Set(val)
+	return a
+}
+
+// Set will set the value to boolValue and will return the previous value
+func (a *AtomicBool) Set(boolValue bool) bool {
+	intValue := int32(falseValue)
+	if boolValue {
+		intValue = trueValue
+	}
+	return a.toTruthy(atomic.SwapInt32(&a.value, intValue))
+}
+
+// Get will return the current value
+func (a *AtomicBool) Get() bool {
+	return a.toTruthy(atomic.LoadInt32(&a.value))
+}
+
+func (a *AtomicBool) toTruthy(val int32) bool {
+	if val == falseValue {
+		return false
+	}
+	return true
```

Comment:
minor thing but you can do `return val == trueValue`

Reply:
also this doesn't need to be on the `AtomicBool` as a method.

Reply:
sounds good, changed slightly to test against false but still single line